src/ngraph/ops/slice.cc

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "ops/slice.h"
+#include "../operator/tensor/matrix_op-inl.h"
+#include "ngraph_sgcompiler_utils.h"
+
+namespace ngraph_bridge {
+
+NgraphNodePtr create_slice_op(const NgraphNodePtr& node,
+                              const nnvm::NodeAttrs& attrs) {
+  const mxnet::op::SliceParam& param =
+      nnvm::get<mxnet::op::SliceParam>(attrs.parsed);
+  nnvm::TShape tshape = NShape_to_TShape(node->get_shape());
+  ngraph::Coordinate ng_begin, ng_end, ng_step;
+  ngraph::AxisSet axes;
+  bool reverse = false;
```

Comment:
Suggestion: IME the algorithm would be slightly clearer if the value of `reverse` was computed just once, early on.  E.g., 

``` c++
...

const bool reverse = (blah blah blah);

for (mxnet::index_t i = 0; i < param.begin.ndim(); ++i) {
   // The 'for' loop as currently written, minus the 'reverse = true;' statement.
   ...
}
```

Reply:
I understand, but `reverse` depends on `step`, which is a tuple. So, we still need to go through the loop.

Reply:
> I understand, but reverse depends on step, which is a tuple. So, we still need to go through the loop.

Right, sorry that my original comment was too hand-wavy to be clear.

I was suggesting that it might be an overall win for readability if we make two passes through the loop: A first pass to determine the value of `reverse`, and then a second pass to do (most of) what you currently do in the loop.

Here's what I meant, in more detail:
``` c++
const bool reverse = std::any_of(param.step.begin(), param.step.end(), 
  [](const dmlc::optional<int> & s) {return s.has_value() && s.value() < 0;});

for (mxnet::index_t i = 0; i < param.begin.ndim(); ++i) {
   ...
}
```